fastlane/Fastfile

```diff
@@ -325,20 +348,55 @@ platform :ios do
 
   desc "Records screenshots for paywalls"
   lane :record_paywall_screenshots do |options|
+    platform = options[:platform] || UI.user_error!("No platform provided. Use one of: ios, iphone-app-on-mac, ipad-app-on-mac, mac-catalyst-optimized-for-mac, mac-catalyst-scaled-to-match-ipad")
+    
+    # Map platform to destination
+    destinations = {
+      "ios" => "platform=iOS Simulator,name=iPhone 16",
+      "iphone-app-on-mac" => "platform=macOS,arch=arm64,variant=Designed for iPhone",
+      "ipad-app-on-mac" => "platform=macOS,arch=arm64,variant=Designed for iPad",
+      "mac-catalyst-optimized-for-mac" => "platform=macOS,arch=arm64,variant=Mac Catalyst",
+      "mac-catalyst-scaled-to-match-ipad" => "platform=macOS,arch=arm64,variant=Mac Catalyst"
+    }
+    
+    # Map platform to TARGETED_DEVICE_FAMILY build setting. This overrides the setting in the Xcode Project,
+    # if we ever change the list of supported platforms we'll need to make sure to update this as well.
+    targeted_device_families = {
+      "mac-catalyst-optimized-for-mac" => "1,2,3,4,6",
+      "mac-catalyst-scaled-to-match-ipad" => "1,2,3,4"
+    }
```

Comment:
I'm seeing [here](https://developer.apple.com/documentation/xcode/build-settings-reference) that `3` is for Apple TV and `4` is for Apple watch. Why are these included in the targeted device families?

Reply:
I included those platforms because they were included in the target before this change, and so I was trying not to mess with anything that was already in there.

Ideally the logic here would be to just add/remove that 6 from the list of targets that was included in the build settings already. But it seemed like a much bigger change to read the build settings here and modify them, than to just pass in a hard-coded override. So I felt like it was a better approach to simply include the full list as it was before this change.